src/Lexer/Lexer.php

```diff
@@ -160,6 +163,7 @@ private function generateRegexp(): string
 			self::TOKEN_CLOSE_CURLY_BRACKET => '\\}',
 
 			self::TOKEN_COMMA => ',',
+			self::TOKEN_COMMENT => '((?<![:/])\/\/[^\n]*|\/\*(?!\*)[^*]*\*+([^\/][^*]*\*+)*\/)',
```

Comment:
How exactly this huge regex works? Is there a danger of it consuming more content than it should?

Reply:
Explanation:

1. `((?<![:/])\/\/[^\n]*|\/\*[\s\S]*?\*\/)`: This entire regular expression is composed of two main alternatives separated by a pipe |.

1. First alternative: `(?<![:/])\/\/[^\n]*`

* `(?<![:/])`: Negative lookbehind to ensure that `//` is not preceded by `:` or `/`, thus avoiding URLs.
* `\/\/`: Matches the start of a single-line comment `//`.
* `[^\n]*`: Matches any character except a newline `\n`, effectively matching the rest of the comment line until the end.


3. Second alternative: `\/\*[\s\S]*?\*\/`

* `\/\*`: Matches the start of a multi-line comment `/*`.
* `[\s\S]*?`: Non-greedy match for any character, including newlines, zero or more times. This matches the content of the multi-line comment.
* `\*\/`: Matches the end of a multi-line comment `*/`.

> Is there a danger of it consuming more content than it should?

I don't think so, but I guess we won't know for sure until it happens. I'm available to support if it comes up.